crates/nu-command/src/filters/update_cells.rs

```diff
@@ -0,0 +1,265 @@
+            columns,
+            span,
+        }
+        .into_pipeline_data(ctrlc))
+    }
+}
+
+struct UpdateCellIterator {
+    input: PipelineIterator,
+    columns: Option<HashSet<String>>,
+    engine_state: EngineState,
+    stack: Stack,
+    block: Block,
+    span: Span,
+}
+
+impl Iterator for UpdateCellIterator {
+    type Item = Value;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        match self.input.next() {
+            Some(val) => {
+                if let Some(ref cols) = self.columns {
+                    if !val.columns().iter().any(|c| cols.contains(c)) {
+                        return Some(val);
+                    }
+                }
+
+                match val {
+                    Value::Record { vals, cols, span } => Some(Value::Record {
+                        vals: cols
+                            .iter()
+                            .zip(vals.into_iter())
+                            .map(|(col, val)| match &self.columns {
+                                Some(cols) if !cols.contains(col) => val,
+                                _ => process_cell(
+                                    val,
+                                    &self.engine_state,
+                                    &mut self.stack,
+                                    &self.block,
+                                    span,
+                                ),
+                            })
+                            .collect(),
+                        cols,
+                        span,
+                    }),
+                    // FIXME: is stepping into nested lists desired?
+                    Value::List { vals, span } => Some(Value::List {
+                        vals: vals
+                            .into_iter()
+                            .map(|val| {
+                                process_cell(
+                                    val,
+                                    &self.engine_state,
+                                    &mut self.stack,
+                                    &self.block,
+                                    span,
+                                )
+                            })
+                            .collect(),
+                        span,
+                    }),
+                    val => Some(process_cell(
+                        val,
+                        &self.engine_state,
+                        &mut self.stack,
+                        &self.block,
+                        self.span,
+                    )),
+                }
+            }
+            None => None,
+        }
+    }
+}
+
+fn process_cell(
+    val: Value,
+    engine_state: &EngineState,
+    stack: &mut Stack,
+    block: &Block,
+    span: Span,
+) -> Value {
+    if let Some(var) = block.signature.get_positional(0) {
+        if let Some(var_id) = &var.var_id {
+            stack.add_var(*var_id, val.clone());
+        }
+    } else {
+        // FIXME: Do we set the $it variable here?
```

Comment:
I think it's OK to leave this empty.

Reply:
Great, would this still be a todo or should we remove it completely?